### PR TITLE
[Enhancement] Support deepspeed with flexible runner

### PR DESCRIPTION
**Pull request by fanqiNO1**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Deepspeed is supported with flexible runner during the training of vit-large and vit-huge.

The performance of vit-large trained with deepspeed is: accuracy/top1: 85.5760 accuracy/top5: 97.4800

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/mae/benchmarks/vit-huge-p14_8xb128-coslr-50e_in1k_deepspeed-zero3.py
Comment (fangyixiao18) [inline configs/mae/benchmarks/vit-huge-p14_8xb128-coslr-50e_in1k_deepspeed-zero3.py:1]: please rename the file to `vit-huge-p14_8xb128-ds-zero3-coslr-50e_in1k.py`

File: configs/mae/benchmarks/vit-huge-p14_8xb128-coslr-50e_in1k_deepspeed-zero3.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+_base_ = [
```
Comment (fangyixiao18) [inline configs/mae/benchmarks/vit-huge-p14_8xb128-coslr-50e_in1k_deepspeed-zero3.py:1]: better to inherit from file `vit-huge-p14_8xb128-coslr-50e_in1k` and remove unchanged keys

File: configs/mae/benchmarks/vit-large-p16_8xb128-coslr-50e_in1k_deepspeed-zero3.py
Comment (fangyixiao18) [inline configs/mae/benchmarks/vit-large-p16_8xb128-coslr-50e_in1k_deepspeed-zero3.py:1]: the same comments as vit-huge config

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: configs/mae/benchmarks/vit-huge-p14_8xb128-ds-zero3-coslr-50e_in1k.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+_base_ = [
```
Comment (fangyixiao18) [inline configs/mae/benchmarks/vit-huge-p14_8xb128-ds-zero3-coslr-50e_in1k.py:1]:
```suggestion
_base_ = ['./vit-huge-p14_8xb128-coslr-50e_in1k.py']
```
only one file is enough, like https://github.com/open-mmlab/mmpretrain/pull/1675/files

File: configs/mae/benchmarks/vit-large-p16_8xb128-ds-zero3-coslr-50e_in1k.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+_base_ = [
```
Comment (fangyixiao18) [inline configs/mae/benchmarks/vit-large-p16_8xb128-ds-zero3-coslr-50e_in1k.py:1]: the same as above

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @fanqiNO1,

We'd like to express our appreciation for your valuable contributions to the mmpretrain. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!

**Review by fangyixiao18 — APPROVED** [review]

**Pull request merged**